src/turbomind/models/llama/LlamaFfnLayer.cc

```diff
@@ -27,10 +27,15 @@
 namespace turbomind {
 
 template<typename T>
-void LlamaFfnLayer<T>::allocateBuffer(size_t token_num)
+void LlamaFfnLayer<T>::allocateBuffer(size_t                     token_num,
+                                      const LlamaDenseWeight<T>* gating,
+                                      const LlamaDenseWeight<T>* inter)
 {
-    inter_buf_          = (T*)allocator_->reMalloc(inter_buf_, sizeof(T) * token_num * inter_size_, false);
-    gating_buf_         = (T*)allocator_->reMalloc(gating_buf_, sizeof(T) * token_num * inter_size_, false);
+    int sz              = sizeof(T) * token_num * inter_size_;
+    int sz_gate         = (gating->lora_policy == 1 && gating->lora_r > 0) ? sz * 3 : sz;
```

Comment:
The same logic `lora_policy && lora_r > 0` appears everywhere. Can we simply set `lora_policy = false` when `lora_r == 0`?